### ISSUE TITLE
validation_epoch_end or on_train_epoch_end receive extra arguments/data

### ISSUE THREAD

**Issue by congchan** [PyTorchLightning/pytorch-lightning#5554]
#### What is your question?

Basically I need to validate my model after each epoch, but the labels information is a little tricky(mixture of list of tuple of int and string), that it could  not be included in DataLoader. 

however, `validation_epoch_end, validation_step, etc..`  only receives DataLoader output.

Is there any solution? 

I actually dun want to add the raw data into module which make the code difficult to understand.
